Add Analytics AdSense linked settings

nfmohit opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #8047 

## Relevant technical choices

<!-- Please describe your changes. -->

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

zutigrm reviewed (approved):
Thanks @nfmohit nice one! LGTM

Good call for setting default value of `adSenseLinkedLastSyncedAt` to `0` instead of `null`, since it is a timestamp value.

I updated QAB to add another step of verifying that `adsenseLinked` setting is present on view only dashboard

techanvil reviewed (approved):
LGTM!

techanvil reviewed (changes requested):
Hey @nfmohit, I approved the PR but then noticed it's developed a merge conflict, please can you resolve this?

nfmohit commented:
> Hey @nfmohit, I approved the PR but then noticed it's developed a merge conflict, please can you resolve this?

Thank you @techanvil. I just fixed the conflicts. I messed it up initially as I was mistakenly in an outdated version of `develop` locally, but I fixed it again. Please take a look, thanks!

techanvil reviewed (approved):
Thanks @nfmohit! LGTM.
